### PR TITLE
gh-107906 Remove unused `maxsize` argument from `_init` method of `queue.Queue`

### LINKED ISSUE #107906: Unused `maxsize` argument in `_init` method of `queue.Queue`
# Feature or enhancement

Note: this is less of a feature or enhancement, and more of a technical debt refactor - however, I don't see an Issue option for "Refactoring," so this is the closest label.

The `__init__` method of `queue.Queue` passes in the `maxsize` argument to the `_init` method:
https://github.com/python/cpython/blob/ee40b3e20d9b8d62a9b36b777dff42db1e9049d5/Lib/queue.py#L34-L36

but the `_init` method does not actually use the `maxsize` argument at all:
https://github.com/python/cpython/blob/ee40b3e20d9b8d62a9b36b777dff42db1e9049d5/Lib/queue.py#L205-L209

# Pitch

Remove this unused `maxsize` argument.

### Linked PRs
* gh-107907

### PR DESCRIPTION
**Pull request by shailshouryya**
Closes #107906

See message in commit bacae73ca5e2f62694c1d2f7875ffc155035e651 for details.

* Issue: gh-107906

### CODE REVIEW

**Comment by shailshouryya** [comment]
I think this PR could skip the news entry since this is just a refactoring change - or would this still need an entry?

**Comment by shailshouryya** [comment]
This shouldn't have broken anything, but it looks like some seemingly unrelated things were passing in the useless `maxsize` argument: https://github.com/python/cpython/actions/runs/5845780591/job/15850226459?pr=107907#step:19:113

**Comment by shailshouryya** [comment]
Taking a look at https://github.com/urllib3/urllib3 to see if this can be updated - will post here again once that is resolved.

**Comment by shailshouryya** [comment]
It looks like `PriorityQueue`
https://github.com/python/cpython/blob/ee40b3e20d9b8d62a9b36b777dff42db1e9049d5/Lib/queue.py#L229-L232

and `LifoQueue`
https://github.com/python/cpython/blob/ee40b3e20d9b8d62a9b36b777dff42db1e9049d5/Lib/queue.py#L245-L248

inherited this `maxsize` argument and their corresponding tests also need to be updated (current failures: https://github.com/python/cpython/actions/runs/5845780591/job/15850226050?pr=107907#step:8:1056)

**Comment by sunmy2019** [comment]
I think you should also take care of `asyncio.queues.Queue` and its subclasses.

**Comment by shailshouryya** [comment]
Following commit e5094abc692b27e9b90aa01727d7be6f87f331cb, all tests [except for the Hypothesis tests on Ubuntu GitHub Action job](https://github.com/python/cpython/actions/runs/5845888855/job/15850444246#step:19:118) now pass. I'm not sure why this specific test is failing, since the error message
```
  File "/home/runner/work/cpython/cpython-ro-srcdir/Lib/queue.py", line 36, in __init__
    self._init()
TypeError: LifoQueue._init() missing 1 required positional argument: '_'
```
points to https://github.com/python/cpython/blob/ee40b3e20d9b8d62a9b36b777dff42db1e9049d5/Lib/queue.py#L34-L36

... but commits bacae73ca5e2f62694c1d2f7875ffc155035e651 and e5094abc692b27e9b90aa01727d7be6f87f331cb should address this. I also tried looking through the [raw logs of the GitHub Action job](https://pipelines.actions.githubusercontent.com/serviceHosts/0881ab16-74dd-4e7f-b53a-5e2dd56508fa/_apis/pipelines/1/runs/189868/signedlogcontent/27?urlExpires=2023-08-13T07%3A29%3A47.2913599Z&urlSigningMethod=HMACV1&urlSignature=T3ed1nJ8kbGTqvTlw%2F%2B23LX3L1xWlmkxhYZHUT6yqxM%3D), but nothing obvious seemed to pop out to me suggesting the test failed to build properly, which I initially suspected to be causing the test failure. (I'm not familiar with the build process for the GitHub Action jobs, though, so I might have missed something.)

File: Lib/asyncio/queues.py [hunk 1]
```diff
@@ -44,7 +44,7 @@ def __init__(self, maxsize=0):
```
Comment (sunmy2019) [inline Lib/asyncio/queues.py:44]: You did not change the call site.

Comment (shailshouryya) [reply Lib/asyncio/queues.py:44]: I missed this, thanks! Updated in commit 8b3df393ec241c11210bf54f5c5669c3ad1b5796

**Comment by hugovk** [comment]
> I think this PR could skip the news entry since this is just a refactoring change - or would this still need an entry?

This needs a news entry because it's changing code which is potentially used.

(There's discussion in the issue whether this change should be made.)

**Comment by rhettinger** [comment]
Closing for the reasons listed in the associated issue.  This is an unnecessary breaking change that overrides the intentional design decision of the original author (Guido).

**Comment by brandonardenwalli** [comment]
Ok, so I am here after https://github.com/python/cpython/issues/107906#issuecomment-1679776110, and this is little more helpful but still not fully clear to me.

I look to https://github.com/python/cpython/commits/ee40b3e20d9b8d62a9b36b777dff42db1e9049d5/Lib/queue.py then https://github.com/python/cpython/commits/f260e443ac07bc794fc9409096679b94528868e2/Lib/Queue.py?browsing_rename_history=true&new_path=Lib/queue.py&original_branch=ee40b3e20d9b8d62a9b36b777dff42db1e9049d5 and find this https://github.com/python/cpython/commit/9022fceae814a3099d8678c662e541f0caeb0026. In https://github.com/python/cpython/commit/9022fceae814a3099d8678c662e541f0caeb0026 I see

```
	# Initialize a queue object with a given maximum size
	# (If maxsize is <= 0, the maximum size is infinite)
	def init(self, maxsize):
		import thread
		self._init(maxsize)
```

and then look at

```
	def _init(self, maxsize):
		self.maxsize = maxsize
		self.queue = []
```

which to me makes sense so far. However, when I look at the current code https://github.com/python/cpython/blob/ee40b3e20d9b8d62a9b36b777dff42db1e9049d5/Lib/queue.py, I see

```
    def __init__(self, maxsize=0):
        self.maxsize = maxsize
        self._init(maxsize)
```

so the code today no longer do anything with maxsize inside of _init

```
    def _init(self, maxsize):
        self.queue = deque()
```

since __init__ now do the

```
        self.maxsize = maxsize
```

itself. After all this, I am confused about:
1. Why to keep maxsize in _init? If I understand correct based on what I learn so far, code having anything that is not used should be remove to avoid confusing other people. In this code, the maxsize is no used in _init, so to me it makes sense to remove. Also, this to me is obvious cause some confusion, so this would be useul to remove.
2. What is purpose of _init? It to me looks like _init does not really do much that the __init__ can do itself. Also, I read that convention is to make _ (underascore) in front of something if it means to tell people not to use that thing or not to make dependent on that thing with _ so why is this a dependent here?
3. Did meaning of _ change? I wonder this since https://github.com/python/cpython/commit/9022fceae814a3099d8678c662e541f0caeb0026 is show to be committed on on Aug 25, 1992, which is quite an old thing, and possible that meaning from then to now change?

I think I may be looking in something wrong again, so @gvanrossum can you please make explanation if you have some time? It is probably that I should be looking somewhere else and the explanation is obvious, but a newbie like me is having trouble finding it.

Also @shailshouryya I just read about some `hypothesis` issue at https://github.com/python/cpython/issues/107862 that talk of the tests not always being the same, so maybe that is the problem that you may be talk of here?

**Comment by gvanrossum** [comment]
I'm sorry @brandonardenwalli, it looks like you need more mentoring than I can provide. This issue has absolutely nothing to do with the question I answered previously, so my conclusion is that you're likely just flailing. Sorry.

FWIW the `_init()` method is essentially an ad-hoc semi-public API for subclasses to override, so I don't want to change it.

**Comment by brandonardenwalli** [comment]
Ok, thanks for the reply! By the way, when you say this is not related to "question I answered previously" you are referring to the issue from yesterday about the issue of Tier 2 interpreters? Just making double check to make sure.

> FWIW the _init() method is essentially an ad-hoc semi-public API for subclasses to override, so I don't want to change it.

Ok, I think that is useful enough. Thank you!

**Comment by gvanrossum** [comment]
Yes that question. Which you started by quoting. And I don’t recall answering any other questions from you.

**Comment by shailshouryya** [comment]
Thanks for linking that `hypothesis` issue, that was an interesting read! Also, in addition to going through the issues with the "Easy" label here, you might also like https://github.com/codecrafters-io/build-your-own-x and https://github.com/practical-tutorials/project-based-learning if you like doing things hands-on. Alternatively, if you like watching videos, I would recommend going on YouTube and searching for something like `make XYZ in python` (or any other language), or browsing through https://www.infoq.com/presentations/.

Best of luck, and have fun :)

**Pull request closed without merging**